### ISSUE TITLE
Support unix timestamps in microseconds (µs)

### ISSUE THREAD

**Issue by camsaul** [metabase/metabase#1889]
Easy to do and some databases (e.g. BigQuery) use them extensively. I'm sure somebody in the wild stores timestamps this way

**Comment by salsakran** [comment]
Is this blocked on the new metadata ontology, @agilliland  or should we just use the same treatment we have for milliseconds and then collect them all into the new ontology post hoc?

**Comment by agilliland** [comment]
Nothing is blocking this technically.  My vote would be to wait an see that it's truly needed though since it's a bit obscure to be using microseconds and it adds bloat to the metadata choices if it's never used.

**Comment by salsakran** [comment]
Yes. I think this is a good issue to leave open in that it's easy and if anyone actually needs it, it's a 5 line commit.

**Comment by camsaul** [comment]
Ya... definitely not a 5-line commit. Support has to be added to each individual driver as well as the `Field` model and frontend

**Comment by Cidan** [comment]
Hi.

We very, very much need this. We store all of our analytics and user records for 30 million users in microseconds due to the high volume nature of our platform.

You get a vote from me!

**Comment by Cidan** [comment]
Are there any plans on supporting this? It would be of immense use to us.

**Comment by chucklessmith** [comment]
This does appear to be fixed in the current release of 0.33, intentionally or not :) Y'all might be relieved to close it.

**Comment by chucklessmith** [comment]
I need to update my earlier response: I swear this was fixed at some point? But I just got an inquiry about this from a user, and lo and behold my test case was still floating around, and it's not working correctly.

I wouldn't say we *need* microsecond precision by any stretch, but that is how we store a lot of timestamps. Even if the value was truncated, it'd be nice to have something displayed without having to explicitly invoke a CAST() or what-have-you.

**Comment by Matt-Yorkley** [comment]
I'm using Postgres and Rails, and have standard `created_at` and `updated_at` timestamps in this format: `1585986703103285`. It would be really nice if this was supported, even if the microseconds were dropped and the field could just be displayed with a simple date and time...

**Comment by selfeky** [comment]
I have the same issue
@maliayas did you find a solution?

**Comment by maliayas** [comment]
Hi @selfeky. I don't remember if I was interested in this issue :) Did GitHub suggest my nickname upon typing @?

**Comment by maliayas** [comment]
Hi @selfeky. I don't remember if I was interested in this issue :) Did GitHub suggest my nickname upon typing @?

**Comment by maliayas** [comment]
Hi @selfeky. I don't remember if I was interested in this issue :) Did GitHub suggest my nickname upon typing @?

**Comment by selfeky** [comment]
@maliayas yes sorry
I meant @Matt-Yorkley

**Comment by Matt-Yorkley** [comment]
What's your setup @selfeky? I'm doing streaming replication to move the data and fixed it by tweaking some configs.
